src/test/java/org/mockitousage/junitrule/StubbingWarningsJUnitRuleTest.java

```diff
@@ -0,0 +1,257 @@
+
+                    declareStubbingWithArg(mock, "a"); // <-- used
+                    declareStubbingWithArg(mock, "b"); // <-- unused
+
+                    useStubbingWithArg(mock, "a");
+                    useStubbingWithArg(mock, "d"); // <-- arg mismatch
+
+                    throw new AssertionError("x");
+                }
+            }, dummy, new DummyTestCase()).evaluate();
+
+            //then
+            fail();
+        } catch (AssertionError e) {
+            assertEquals("x", e.getMessage());
+            assertEquals(
+                "[MockitoHint] DummyTestCase.null (see javadoc for MockitoHint):\n" +
+                "[MockitoHint] 1. Unused... -> at org.mockitousage.junitrule.StubbingWarningsJUnitRuleTest.declareStubbingWithArg(StubbingWarningsJUnitRuleTest.java:0)\n" +
+                "[MockitoHint]  ...args ok? -> at org.mockitousage.junitrule.StubbingWarningsJUnitRuleTest.useStubbingWithArg(StubbingWarningsJUnitRuleTest.java:0)\n",
+                    filterLineNo(logger.getLoggedInfo()));
+        }
+    }
+
+    @Test
+    public void no_mismatch_when_stub_was_used() throws Throwable {
+        try {
+            //when
+            jUnitRule.apply(new Statement() {
+                public void evaluate() throws Throwable {
+                    IMethods mock = mock(IMethods.class);
+
+                    declareStubbingWithArg(mock, "a");
+
+                    useStubbingWithArg(mock, "a");
+                    useStubbingWithArg(mock, "d"); // <-- arg mismatch, but the stub was already used
+
+                    throw new AssertionError("x");
+                }
+            }, dummy, new DummyTestCase()).evaluate();
+
+            //then
+            fail();
+        } catch (AssertionError e) {
+            assertEquals("x", e.getMessage());
+            assertTrue(logger.isEmpty());
+        }
+    }
+
+    @Test
+    public void no_stubbing_arg_mismatch_on_pass() throws Throwable {
+        //given
+        jUnitRule.apply(new Statement() {
+            public void evaluate() throws Throwable {
+                IMethods mock = mock(IMethods.class);
+                declareStubbingWithArg(mock, "a");
+                useStubbingWithArg(mock, "b");
+            }
+        }, dummy, new DummyTestCase()).evaluate();
+
+        //expect
+        assertEquals(
+            "[MockitoHint] DummyTestCase.null (see javadoc for MockitoHint):\n" +
+            "[MockitoHint] 1. Unused -> at org.mockitousage.junitrule.StubbingWarningsJUnitRuleTest.declareStubbingWithArg(StubbingWarningsJUnitRuleTest.java:0)\n",
+                filterLineNo(logger.getLoggedInfo()));
+    }
+
+    @Test
+    public void warns_about_unused_stubs_when_passed() throws Throwable {
+        jUnitRule.apply(new Statement() {
+            public void evaluate() throws Throwable {
+                IMethods mock = mock(IMethods.class);
+                declareStubbing(mock);
+            }
```

Comment:
This anonymous `Statement` should be moved to a nested type or method cause it is duplicated.